paimon-common/src/main/java/org/apache/paimon/io/cache/CachedRandomInputView.java

```diff
@@ -50,12 +52,24 @@ public class CachedRandomInputView extends AbstractPagedInputView
 
     private int currentSegmentIndex;
 
+    /**
+     * The thread pool is involved to solve computeIfAbsent issue. From JDK 9, computeIfAbsent will
+     * throw ConcurrentModificationException if it is detected that the mapping function modifies
+     * this map during computation. So we add the thread pool to remove segments expired page. To
+     * guarantee the concurrency safety, we need to replace the HashMap with ConcurrentHashMap.
+     *
+     * <p>Note: if the cache expires many pages in a while, the single thread executor can't process
+     * it. That means, some remove actions are blocked, this will cause dirty data read. But this
+     * occurs with low probability.
+     */
+    ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
We don't need to introduce this complex codes.
How about just modify `getCurrentPage`.
```
    private MemorySegment getCurrentPage() {
        MemorySegment segment = segments.get(currentSegmentIndex);
        if (segment == null) {
            segment = cacheManager.getPage(file, currentSegmentIndex, this::invalidPage);
            segments.put(currentSegmentIndex, segment);
        }
        return segment;
    }
```

Reply:
ok. 

Actually, this is the first edition of this PR.